contracts/MonetarySupervisor.sol

```diff
@@ -140,8 +149,54 @@ contract MonetarySupervisor is Restricted, TokenReceiver { // solhint-disable-li
     }
 
     // helper function for FrontEnd to reduce calls
-    function getParams() external view returns(uint[2]) {
-        return [ltdDifferenceLimit, allowedLtdDifferenceAmount];
+    function getParams() external view returns(uint[3]) {
+        return [ltdLockDifferenceLimit, ltdLoanDifferenceLimit, allowedLtdDifferenceAmount];
+    }
+
+    function getLoanToDepositRatio() external view returns (uint loanToDepositRatio) {
+        loanToDepositRatio = totalLockedAmount == 0 ? 0 : totalLockedAmount.mul(PERCENT_100).div(totalLoanAmount);
+    }
+
+    /* Helper function for UI.
+        Returns max lock amount based on minLockAmount, interestPt, using LTD params & interestEarnedAccount balance */
+    function getMaxLockAmount(uint minLockAmount, uint interestPt) external view returns (uint maxLock) {
+        uint allowedByEarning = augmintToken.balanceOf(address(interestEarnedAccount)).mul(PERCENT_100).div(interestPt);
+        uint allowedByLtd = getMaxLockAmountAllowedByLtd();
+        maxLock = allowedByEarning < allowedByLtd ? allowedByEarning : allowedByLtd;
+        maxLock = maxLock < minLockAmount ? 0 : maxLock;
+    }
+
+    /* Helper function for UI.
+        Returns max loan amount based on minLoanAmont using LTD params */
+    function getMaxLoanAmount(uint minLoanAmount) external view returns (uint maxLoan) {
+        uint allowedByLtd = getMaxLoanAmountAllowedByLtd();
+        maxLoan = allowedByLtd < minLoanAmount ? 0 : allowedByLtd;
+    }
+
+    /* returns maximum lockable token amount allowed by LTD params. */
+    function getMaxLockAmountAllowedByLtd() public view returns(uint maxLockByLtd) {
+        uint allowedByLtdDifferencePt = totalLoanAmount.mul(PERCENT_100).div(PERCENT_100.sub(ltdLockDifferenceLimit));
+        allowedByLtdDifferencePt = totalLockedAmount >= allowedByLtdDifferencePt ?
+                                        0 : allowedByLtdDifferencePt.sub(totalLockedAmount);
+
+        uint allowedByLtdDifferenceAmount = totalLoanAmount > totalLockedAmount.add(allowedLtdDifferenceAmount) ?
+                                        0 : totalLoanAmount.add(allowedLtdDifferenceAmount).sub(totalLockedAmount);
+
+        maxLockByLtd = allowedByLtdDifferencePt > allowedByLtdDifferenceAmount ?
+                                        allowedByLtdDifferencePt : allowedByLtdDifferenceAmount;
+    }
+
+    /* returns maximum borrowable token amount allowed by LTD params */
+    function getMaxLoanAmountAllowedByLtd() public view returns(uint maxLoanByLtd) {
```

Comment:
`getMaxLoan*` and `getMaxLock*` are basically identical. refactor to a single function.

Reply:
not sure I got you.
getMaxLockAmount is an 'extension' of getMaxLockAmountAllowedByLtd 
getMaxLockAmount is a helper function for UI (when selected lock product is known)
getMaxLockAmountAllowedByLtd is used internally by requestInterest() .
we could make internal fx (minor benefits) but unit tests are testing them at the moment.

Reply:
the point is that there's basically copy-pasted code shared by the two methods. you should extract this part into a common method. something along these lines:

```
    function max(uint allowedByLtdDifferencePt) internal returns(uint max) {
        allowedByLtdDifferencePt = totalLockedAmount >= allowedByLtdDifferencePt ?
                                        0 : allowedByLtdDifferencePt.sub(totalLockedAmount);

        uint allowedByLtdDifferenceAmount = totalLoanAmount > totalLockedAmount.add(allowedLtdDifferenceAmount) ?
                                        0 : totalLoanAmount.add(allowedLtdDifferenceAmount).sub(totalLockedAmount);

        return allowedByLtdDifferencePt > allowedByLtdDifferenceAmount ?
                                        allowedByLtdDifferencePt : allowedByLtdDifferenceAmount;

    }
```

Reply:
ok got you. but the first line is not the same.  But indeed, this part can be moved to a generic fx:
```js
uint allowedByLtdDifferenceAmount = totalLoanAmount > totalLockedAmount.add(allowedLtdDifferenceAmount) ?
            0 : totalLockedAmount.add(allowedLtdDifferenceAmount).sub(totalLoanAmount);

maxLoanByLtd = allowedByLtdDifferencePt > allowedByLtdDifferenceAmount ?
             allowedByLtdDifferencePt : allowedByLtdDifferenceAmount;
```

Reply:
well even that part is not the same :) 
can't figure a concise & readable way to make it a common function. Leaving it as unless you have a better idea